### PR TITLE
Stack and queue

### PR DESCRIPTION
**Pull request by MerrybyPractice**

### CODE REVIEW

**Comment by MerrybyPractice** [comment]
- [x] Create a Node class that has properties for the value stored in the Node, and a pointer to the next node.
- [x] Create a Stack class that has a top property. It creates an empty Stack when instantiated.
- [x] This object should be aware of a default empty value assigned to top when the stack is created.
- [x] Define a method called push which takes any value as an argument and adds a new node with that value to the top of the stack with an O(1) Time performance.
- [x] Define a method called pop that does not take any argument, removes the node from the top of the stack, and returns the node’s value.
- [x] Define a method called peek that does not take an argument and returns the value of the node located on the top of the stack.
- [x] Create a Queue class that has a top property. It creates an empty queue when instantiated.
- [x] This object should be aware of a default empty value assigned to front when the queue is created.
- [x] Define a method called enqueue which takes any value as an argument and adds a new node with that value to the back of the queue with an O(1) Time performance.
- [x] Define a method called dequeue that does not take any argument, removes the node from the front of the queue, and returns the node’s value.
- [x] Define a method called peek that does not take an argument and returns the value of the node located in the front of the stack.
- [x] At no time should an exception or stack trace be shown to the end user. Catch and handle any such exceptions and return a printed value or operation which cleanly represents the state and either stops execution cleanly, or provides the user with clear direction and output.
- [x] Be sure to follow your languages best practices for naming conventions.
- [x] You have access to the Node class and all the properties on the Linked List class.
Tests: 
- [x] Can successfully push onto a stack
- [x] Can successfully push multiple nodes onto a stack
- [x] Can successfully pop off the stack
- [x] Can successfully empty a stack after multiple pops.
- [x] Can successfully peek the next item on the stack.
- [x] Can successfully instantiate an empty stack
- [x] Can successfully enqueue onto a queue
- [x] Can successfully enqueue multiple items into a queue
- [x] Can successfully dequeue off of a queue the expected value
- [x] Can successfully peek into a queue, seeing the expected value
- [x] Can successfully empty a queue after multiple dequeues
- [x] Can successfully instantiate an empty queue

File: src/test/java/stacksandqueues/QueueTest.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+public class QueueTest {
+    //enqueue multiple items
+    @Test
+    public void testEnqueue() {
+        Queue testQueue = new Queue();
+        testQueue.enqueue(1);
+        testQueue.enqueue(2);
+        testQueue.enqueue(3);
+
```
Comment (JessLovell) [inline src/test/java/stacksandqueues/QueueTest.java:15]: You could also have assert statements between each enqueue to verify that it is doing what is expected. As well as dequeue something then enqueue it to ensure that it's right. 

What happens when DQed to null then enqueued?

File: src/test/java/stacksandqueues/QueueTest.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        assertEquals(1, testQueue.dequeue());
+        assertEquals(2, testQueue.front.value);
+        assertEquals(2, testQueue.dequeue());
+        assertEquals(3, testQueue.dequeue());
+        assertNull(testQueue.front);
+        assertNull(testQueue.rear);
```
Comment (JessLovell) [inline src/test/java/stacksandqueues/QueueTest.java:34]: Throwing some EnQ's in between to make sure it behaves as expected.

File: src/test/java/stacksandqueues/StackTest.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
+package stacksandqueues;
+
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+public class StackTest {
+
+    //push multiple nodes on to stack
```
Comment (JessLovell) [inline src/test/java/stacksandqueues/StackTest.java:9]: No test for peeking? 😢

File: src/test/java/stacksandqueues/QueueTest.java [hunk 4]
```diff
@@ -0,0 +1,65 @@
+package stacksandqueues;
+
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+public class QueueTest {
+    //enqueue multiple items
```
Comment (JessLovell) [inline src/test/java/stacksandqueues/QueueTest.java:8]: Test for emptying a queue after multiple dequeues

**Review by JessLovell — COMMENTED** [review]
Great Job Merry! Just a few comments on testing.

**Pull request merged**